duplicacy-autobackup.sh

```diff
@@ -24,24 +24,34 @@ do_init() {
 
 do_backup() {
   status=0
-  if [[ -f $PRE_BACKUP_SCRIPT ]]; then
-    echo "Running pre-backup script"
-    sh $PRE_BACKUP_SCRIPT
-    status=$?
-  fi
-  if [[ $status != 0 ]]; then
-    echo "Pre-backup script exited with status code $status. Not performing backup." >&2
-    return
-  fi
+  while : ; do
+    if [[ -f $PRE_BACKUP_SCRIPT ]]; then
+      echo "Running pre-backup script"
+      sh $PRE_BACKUP_SCRIPT
+      status=$?
+      if [[ $status != 0 ]]; then
+        echo "Pre-backup script exited with status code $status. Not performing backup." >&2
+        break
+      fi
+    fi
 
-  duplicacy backup $DUPLICACY_BACKUP_OPTIONS
-
-  if [[ -f $POST_BACKUP_SCRIPT ]]; then
-    echo "Running post-backup script"
-    sh $POST_BACKUP_SCRIPT
+    duplicacy backup $DUPLICACY_BACKUP_OPTIONS
     status=$?
-    echo "Post-backup script exited with status $status"
-  fi
+    if [[ $status != 0 ]]; then
```

Comment:
Yup, that can make sense

Reply:
It's a simple way to break out of the `do_backup` function on error.
Other options are to return early or use a bunch of nested `if` statements.

Reply:
Personally I think returning early is easier to understand